### PR TITLE
feat: add ends at env variable

### PR DESCRIPTION
**Pull request by vimwitch**

### CODE REVIEW

File: packages/frontend/src/pages/Contribute.jsx [hunk 1]
```diff
@@ -467,11 +467,7 @@ export default observer(() => {
                             fontWeight: '600',
                           }}
                           onClick={async () => {
-                            try {
-                              await ceremony.join(name, 'open')
-                            } catch (e) {
-                              setError(e)
-                            }
+                            await ceremony.join(name, 'open')
```
Comment (kittybest) [inline packages/frontend/src/pages/Contribute.jsx:470]: Why remove the try&catch here?

File: packages/backend/src/config.mjs [hunk 2]
```diff
@@ -33,6 +33,8 @@ export const CONTRIBUTION_TIMEOUT = +(
 )
 export const PRUNE_INTERVAL = +(process.env.PRUNE_INTERVAL ?? 12 * 1000)
 
+export const ENDS_AT = +(process.env.ENDS_AT ?? 10 ** 13)
```
Comment (kittybest) [inline packages/backend/src/config.mjs:36]: Why `10**13` here? or it just a random number?
For the timestamp for Dec 12 2023 here ⬇️
image

File: packages/frontend/src/pages/Contribute.jsx (hunk 1 above)
Comment (vimwitch) [reply packages/frontend/src/pages/Contribute.jsx:470]: The [`Button`](https://github.com/Unirep/ceremony/blob/main/packages/frontend/src/components/Button.jsx#L7) component will catch the error and display the message. The `setError` handler in this component doesn't properly handle `Error` objects, the check for `error.length` always returns `undefined` in the hook.

File: packages/backend/src/config.mjs (hunk 2 above)
Comment (kittybest) [reply packages/backend/src/config.mjs:36]: Also use this in frontend as default `ENDS_AT` timestamp

**Review by kittybest — APPROVED** [review]

**Pull request merged**